sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobItem.java

```diff
@@ -48,6 +49,24 @@ public BlobItem(BlobItemInternal blobItemInternal) {
                 this.tags.put(tag.getKey(), tag.getValue());
             }
         }
+        this.objectReplicationSourcePolicies = null;
+        if (blobItemInternal.getObjectReplicationMetadata() != null) {
+            this.objectReplicationSourcePolicies = new HashMap<>();
+            for (String orString : blobItemInternal.getObjectReplicationMetadata().keySet()) {
```

Comment:
I think CI will tell you this, but we should use entrySet instead of keySet so we don't have to index into the map

Reply:
done